dropwizard-logging/src/main/java/io/dropwizard/logging/LoggingFactory.java

```diff
@@ -110,6 +126,8 @@ public void configure(MetricRegistry metricRegistry, String name) {
             root.addAppender(output.build(root.getLoggerContext(), name, null));
         }
 
+        printIfErrorsOccured(root.getLoggerContext());
```

Comment:
I'm more fond to logging to the error output stream. I believe it's more suitable for this kind of messages.

We can redirect `StatusPrinter` to `System.err`, but also should have bear in mind that `StatusPrinter` for some reason is static, so we should restore it to `System.out` after using. How about making the output stream as a parameter of `LoggingFactory`? Something like that:

````java
@JsonIgnore                                         
private final PrintStream configurationErrorsStream;
...
@VisibleForTesting
LoggingFactory(LoggerContext loggerContext, PrintStream configurationErrorsStream) {
    this.loggerContext = checkNotNull(loggerContext);
    this.errorsStream = checkNotNull(configurationErrorsStream);
}
...

StatusPrinter.setPrintStream(configurationErrorsStream);
try {
    printIfErrorsOccured(loggerContext);
}  finally {
   StatusPrinter.setPrintStream(System.out); 
}
````

As a benefit, we can pass a fake stream in the test to the class in the constructor.